src/modules/assets/components/RequestAsset.vue

```diff
@@ -0,0 +1,395 @@
+<script>
+import BaseAlert from "@/components/BaseAlert.vue";
+import { requestAsset } from "../services";
+
+export default {
+  name: "RequestAsset",
+  data() {
+    return {
+      name: {
+        value: "",
+        error: "",
+      },
+      category: {
+        value: "select a category",
+      },
+      isNewProduct: {
+        value:false,
+      },
+      submission: {
+        message: "",
+        isVerified: false,
+      },
+    };
+  },
+  components: {
+    BaseAlert,
+  },
+  methods: {
+    formData() {
+      return {
+        name: this.name.value,
+        category: this.category.value,
+        isNewProduct: this.isNewProduct.value,
+      };
+    },
+    validateField(field) {
+      console.log('entered func');
+      if (field === "name") {
+        if (this.name.value.length <  3){
+          this.name.error = "asset name should be more than three letters long";
+          console.log(this.name.value.length);
+        }
+        else{
+        this.name.error = ""
+      }
+      }
+      
+    },
+    async handleSubmit() {
+      if (!this.name.value ) {
+        return (this.submission.message = "Field must not be empty");
+      }
+      if (this.name.error ) {
+        return (this.submission.message =
+          "Some fields are not filled properly");
+      }
+      if(this.category.value === "select a category"){
+        return (this.submission.message =
+          "Please select a category");
+      }
+      this.submission.message = "Sorry, you cannot request an asset right now"
+      // try {
```

Comment:
If the commented code is not required, please remove it.

Reply:
The commented code is required when backend is ready for integration

Reply:
OK then.